extension/sql/meta.sql

```diff
@@ -20,6 +20,39 @@ GRANT SELECT ON ALL SEQUENCES IN SCHEMA vectorize TO pg_monitor;
 ALTER DEFAULT PRIVILEGES IN SCHEMA vectorize GRANT SELECT ON TABLES TO pg_monitor;
 ALTER DEFAULT PRIVILEGES IN SCHEMA vectorize GRANT SELECT ON SEQUENCES TO pg_monitor;
 
+CREATE OR REPLACE FUNCTION handle_table_drop()
+RETURNS event_trigger AS $$
+DECLARE
+    obj RECORD;
+    schema_name TEXT;
+    table_name TEXT;
+BEGIN
+    FOR obj IN SELECT * FROM pg_event_trigger_dropped_objects() LOOP
+        IF obj.object_type = 'table' THEN
+            schema_name := split_part(obj.object_identity, '.', 1);  
+            table_name := split_part(obj.object_identity, '.', 2);  
+            
+            RAISE NOTICE 'Event Trigger Fired for table drop: %, schema: %', table_name, schema_name;
+            
+            -- Perform cleanup: delete the associated job from the vectorize.job table
+            DELETE FROM vectorize.job
+            WHERE params ->> 'table' = table_name
+            AND params ->> 'schema' = schema_name;
+            
+            IF NOT FOUND THEN
+                RAISE NOTICE 'No matching job found for table: %, schema: %', table_name, schema_name;
+            END IF;
```

Comment:
Will this notice fire on all `DROP TABLE` in the database?

Reply:
I've used the RAISE NOTICE statements for debugging purposes to confirm that the event trigger is firing correctly. However, in production, these notices are not necessary, so I'll remove them.

Reply:
Done.